apps/graphql/test/mutations/shell_mutations_test.exs

```diff
@@ -52,6 +52,86 @@ defmodule GraphQl.ShellMutationsTest do
     end
   end
 
+  describe "updateShell" do
+    test "it will update an existing shell instance" do
+      %{email: e} = user = insert(:user, roles: %{admin: true})
+
+      expect(Pods, :fetch, fn _ -> {:ok, Pods.pod("plrl-shell-2", e)} end)
+      expect(Core.Shell.Scm, :setup_repository, fn
+        :github, ^e, "tok", nil, "demo" ->
+          {:ok, "git@github.com:pluralsh/demo.git", "pub-key", "priv-key", nil}
+      end)
+
+      expect(Core.Clients.Vault, :write, fn _, _ -> {:ok, %{}} end)
+
+      attrs = %{
+        "provider" => "AWS",
+        "scm" => %{"provider" => "GITHUB", "token" => "tok", "name" => "demo"},
+        "workspace" => %{
+          "cluster" => "plural",
+          "region" => "us-east-1",
+          "bucketPrefix" => "plrl",
+          "subdomain" => "demo.onplural.sh"
+        },
+        "credentials" => %{
+          "aws" => %{
+            "accessKeyId" => "access-key",
+            "secretAccessKey" => "secret-key"
+          }
+        }
+      }
+
+      {:ok, %{data: %{"createShell" => created}}} = run_query("""
+        mutation Create($attrs: CloudShellAttributes!) {
+          createShell(attributes: $attrs) {
+            id
+            provider
+            gitUrl
+            aesKey
+            updatedAt
+          }
+        }
+      """, %{"attrs" => attrs}, %{current_user: user})
+
+      assert created["id"]
+      assert created["provider"] == "AWS"
+      assert created["gitUrl"] == "git@github.com:pluralsh/demo.git"
+      assert created["aesKey"]
+      assert created["updatedAt"]
+
+      expect(Kazan, :run, fn _ -> {:ok, %{}} end)
+
+      attrs = %{
+        "provider" => "GCP",
+        "scm" => %{"provider" => "GITHUB", "token" => "tok", "name" => "demo"},
+        "workspace" => %{
+          "cluster" => "plural",
+          "region" => "us-east-2",
+          "bucketPrefix" => "plrl",
+          "subdomain" => "demo.onplural.sh"
+        },
+        "credentials" => %{
```

Comment:
fwiw, this is the only block that's honored by the update

Reply:
I probably would have written this test by using the `insert(:cloud_shell)` helper like in some other tests, but this is fine too

Reply:
> fwiw, this is the only block that's honored by the update

oh yeah, I was wondering if this is what this `update_changeset` does.
```
  def update_changeset(model, attrs \\ %{}) do
    model
    |> cast(attrs, [])
    |> cast_embed(:credentials)
  end
```